File: Java/SpaceVIL/spacevil/sandbox/src/main/java/sandbox/View/VisualContact.java
package sandbox.View;

import java.awt.Color;

import com.spvessel.spacevil.*;
import com.spvessel.spacevil.Core.IMouseMethodState;
import com.spvessel.spacevil.Core.Position;
import com.spvessel.spacevil.Decorations.Figure;
import com.spvessel.spacevil.Decorations.ItemState;
import com.spvessel.spacevil.Decorations.Shadow;
import com.spvessel.spacevil.Flags.ItemAlignment;
import com.spvessel.spacevil.Flags.ItemStateType;
import com.spvessel.spacevil.Flags.SizePolicy;

public class VisualContact extends Prototype {
    public static int _count = 0;

    // private Integer[] _array;

    public VisualContact() {
        // self view attr
        setItemName("VC_" + _count);
        setBackground(80, 80, 80);
        setMinSize(250, 60);
        setSize(300, 60);
        setSizePolicy(SizePolicy.Expand, SizePolicy.Fixed);
        setAlignment(ItemAlignment.Top, ItemAlignment.Left);
        setBorderRadius(15);
        ItemState hover = new ItemState();
        hover.background = new Color(255, 255, 255, 15);
        addItemState(ItemStateType.Hovered, hover);
        setPadding(10, 0, 5, 0);
        setMargin(1, 1, 1, 1);
        effects().add(new Shadow(10, new Position(3, 3), new Color(0, 0, 0, 160)));
        _count++;
    }

    @Override
    public void initElements() {
        // contact image border
        Ellipse border = new Ellipse();
        border.setBackground(255, 180, 100, 100);
        border.setHeight(45);
        border.setHeightPolicy(SizePolicy.Fixed);
        border.setWidth(45);
        border.setWidthPolicy(SizePolicy.Fixed);
        border.setAlignment(ItemAlignment.VCenter, ItemAlignment.Left);
        border.quality = 32;
        // border.setTriangles(GraphicsMathService.getRoundSquare(50, 50, 25, 0, 0));
        // border.setTriangles(GraphicsMathService.getEllipse(50, 50, 0, 0, 32));

        // contact name
        Label name = new Label(getItemName() + " contact");
        // name.isHover = false;
        // Label name = new Label();
        name.setTextAlignment(ItemAlignment.VCenter, ItemAlignment.Left);
        name.setBackground(255, 255, 255, 32);
        name.setForeground(210, 210, 210);
        name.setHeight(30);
        name.setHeightPolicy(SizePolicy.Fixed);
        name.setWidthPolicy(SizePolicy.Expand);
        // name.setMargin(60, 0, 30, 5);
        name.setPadding(20, 0, 0, 0);
        name.setAlignment(ItemAlignment.Bottom, ItemAlignment.Left);
        name.setBorderRadius(10);
        name.addItemState(ItemStateType.Hovered, new ItemState(new Color(10, 162, 232)));

        // contact close
        ButtonCore close = new ButtonCore();
        close.setBackground(40, 40, 40);
        close.setWidth(14);
        close.setWidthPolicy(SizePolicy.Fixed);
        close.setHeight(14);
        close.setHeightPolicy(SizePolicy.Fixed);
        close.setAlignment(ItemAlignment.Top, ItemAlignment.Right);
        // close.setMargin(0, 5, 0, 0);
        close.setCustomFigure(new Figure(false, GraphicsMathService.getCross(14, 14, 5, 45)));
        ItemState hover = new ItemState();
        hover.background = new Color(255, 255, 255, 125);
        close.addItemState(ItemStateType.Hovered, hover);
        IMouseMethodState click = (sender, args) -> disposeSelf();
        close.eventMouseClick.add(click);

        // adding
        addItem(border);
        addItem(name);
        addItem(close);

        // _array = new Integer[1024 * 1024 * 128];
        // for (int i = 0; i < _array.length; i++) {
        // _array[i] = 12;
        // }
    }

    public void disposeSelf() {
        getParent().removeItem(this);
    }
}

File: sandbox/View/Components/VisualContact.cs
using SpaceVIL;
using System.Drawing;
using SpaceVIL.Core;
using SpaceVIL.Decorations;
using View.Decorations;

namespace View.Components
{
    class VisualContact : Prototype
    {
        private Label contactName = null;
        public VisualContact(string name)
        {
            SetBackground(80, 80, 80);
            SetSize(300, 60);
            SetAlignment(ItemAlignment.Top, ItemAlignment.Left);
            SetSizePolicy(SizePolicy.Expand, SizePolicy.Fixed);
            SetHeight(80);
            SetBorderRadius(10);
            ItemState hover = new ItemState();
            hover.Background = Color.FromArgb(15, 255, 255, 255);
            AddItemState(ItemStateType.Hovered, hover);
            SetPadding(10, 5, 5, 5);
            SetMargin(1, 1, 1, 1);
            Effects().Add(new Shadow(10, new Position(3, 3), Palette.BlackShadow));

            contactName = new Label(name);
        }

        public override void InitElements()
        {
            Ellipse contactPhoto = new Ellipse();
            contactPhoto.SetBackground(Palette.GreenLight);
            contactPhoto.SetSizePolicy(SizePolicy.Fixed, SizePolicy.Fixed);
            contactPhoto.SetSize(70, 70);
            contactPhoto.SetAlignment(ItemAlignment.VCenter, ItemAlignment.Left);
            contactPhoto.Quality = 32;
            contactPhoto.Effects().Add(new Shadow(10, new Position(3, 3), Palette.BlackShadow));

            // contact name
            contactName.SetTextAlignment(ItemAlignment.VCenter, ItemAlignment.Left);
            contactName.SetFontSize(16);
            contactName.SetFontStyle(FontStyle.Bold);
            contactName.SetBackground(255, 255, 255, 32);
            contactName.SetForeground(210, 210, 210);
            contactName.SetHeight(40);
            contactName.SetHeightPolicy(SizePolicy.Fixed);
            contactName.SetWidthPolicy(SizePolicy.Expand);
            contactName.SetPadding(20, 0, 0, 0);
            contactName.SetMargin(80, 0, 15, 10);
            contactName.SetAlignment(ItemAlignment.Bottom, ItemAlignment.Left);
            contactName.SetBorderRadius(20);

            // contact close
            ButtonCore removeContactBtn = new ButtonCore();
            removeContactBtn.SetBackground(40, 40, 40);
            removeContactBtn.SetWidth(14);
            removeContactBtn.SetWidthPolicy(SizePolicy.Fixed);
            removeContactBtn.SetHeight(14);
            removeContactBtn.SetHeightPolicy(SizePolicy.Fixed);
            removeContactBtn.SetAlignment(ItemAlignment.Top, ItemAlignment.Right);
            removeContactBtn.SetMargin(0, 5, 0, 0);
            removeContactBtn.SetCustomFigure(new Figure(false, GraphicsMathService.GetCross(14, 14, 4, 45)));
            removeContactBtn.AddItemState(ItemStateType.Hovered, new ItemState(Palette.WhiteGlass));
            removeContactBtn.EventMouseClick += (sender, args) => DisposeSelf();

            AddItems(contactPhoto, contactName, removeContactBtn);
        }

        public void DisposeSelf()
        {
            GetParent().RemoveItem(this);
        }
    }
}
